src/expr/src/expr/expr_unary.rs

```diff
@@ -102,6 +102,12 @@ macro_rules! gen_cast {
             { varchar, boolean, str_to_bool },
 
             { boolean, varchar, bool_to_str },
+            { boolean, int16, general_cast },
+            { boolean, int32, general_cast },
+            { boolean, int64, general_cast },
+            { int16, boolean, int16_to_bool },
+            { int32, boolean, int32_to_bool },
+            { int64, boolean, int64_to_bool },
```

Comment:
PostgreSQL only allows casting between boolean and int32. We do not need to use macro/generic to support int16 and int64.